pkg/resources/tag_association.go

```diff
@@ -101,132 +104,166 @@ func TagAssociation() *schema.Resource {
 	}
 }
 
-// CreateTagAssociation implements schema.CreateFunc.
-func CreateTagAssociation(d *schema.ResourceData, meta interface{}) error {
-	db := meta.(*sql.DB)
-	tagID := d.Get("tag_id").(string)
-	objectType := d.Get("object_type").(string)
-	tagValue := d.Get("tag_value").(string)
-	objectDatabase, objectSchema, objectName := expandObjectIdentifier(d.Get("object_identifier"))
-	fullyQualifierObjectIdentifier := snowflakeValidation.FormatFullyQualifiedObjectID(objectDatabase, objectSchema, objectName)
-
-	builder := snowflake.NewTagAssociationBuilder(tagID).WithObjectIdentifier(fullyQualifierObjectIdentifier).WithObjectType(objectType).WithTagValue(tagValue)
-
-	q := builder.Create()
-	if err := snowflake.Exec(db, q); err != nil {
-		return fmt.Errorf("error associating tag to object: [%v] with command: [%v], tag_id [%v]", fullyQualifierObjectIdentifier, q, tagID)
+func useTagDatabaseAndSchema(ot sdk.ObjectType) bool {
+	switch ot {
+	case sdk.ObjectTypeColumn, sdk.ObjectTypePipe, sdk.ObjectTypeStream, sdk.ObjectTypeStage, sdk.ObjectTypeTable, sdk.ObjectTypeTask, sdk.ObjectTypeView:
+		return true
+	default:
+		return false
 	}
+}
 
-	skipValidate := d.Get("skip_validation").(bool)
-	if !skipValidate {
-		log.Println("[DEBUG] validating tag creation")
+func tagIdentifierAndObjectIdentifier(d *schema.ResourceData) (sdk.SchemaObjectIdentifier, []sdk.ObjectIdentifier, sdk.ObjectType) {
```

Comment:
This method is complicated and have multiple possible outputs. Please add unit test and validate different possibilities.

Reply:
added unit tests